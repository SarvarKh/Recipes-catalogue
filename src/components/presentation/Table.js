import PropTypes from 'prop-types';

const Table = ({ detailMeal }) => (
  <table>
    <tbody>
      <tr>
        <th>No</th>
        <th>Ingredients</th>
        <th>Measure</th>
      </tr>
      {detailMeal.strIngredient1
        ? (
          <tr>
            <td>1</td>
            <td>{detailMeal.strIngredient1}</td>
            <td>{detailMeal.strMeasure1}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient2
        ? (
          <tr>
            <td>2</td>
            <td>{detailMeal.strIngredient2}</td>
            <td>{detailMeal.strMeasure2}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient3
        ? (
          <tr>
            <td>3</td>
            <td>{detailMeal.strIngredient3}</td>
            <td>{detailMeal.strMeasure3}</td>
          </tr>
        )
        : null}

      {detailMeal.strIngredient4
        ? (
          <tr>
            <td>4</td>
            <td>{detailMeal.strIngredient4}</td>
            <td>{detailMeal.strMeasure4}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient5
        ? (
          <tr>
            <td>5</td>
            <td>{detailMeal.strIngredient5}</td>
            <td>{detailMeal.strMeasure5}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient6
        ? (
          <tr>
            <td>6</td>
            <td>{detailMeal.strIngredient6}</td>
            <td>{detailMeal.strMeasure6}</td>
          </tr>
        )
        : null}

      {detailMeal.strIngredient7
        ? (
          <tr>
            <td>7</td>
            <td>{detailMeal.strIngredient7}</td>
            <td>{detailMeal.strMeasure7}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient8
        ? (
          <tr>
            <td>8</td>
            <td>{detailMeal.strIngredient8}</td>
            <td>{detailMeal.strMeasure8}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient9
        ? (
          <tr>
            <td>9</td>
            <td>{detailMeal.strIngredient9}</td>
            <td>{detailMeal.strMeasure9}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient10
        ? (
          <tr>
            <td>10</td>
            <td>{detailMeal.strIngredient10}</td>
            <td>{detailMeal.strMeasure10}</td>
          </tr>
        )
        : null}

      {detailMeal.strIngredient11
        ? (
          <tr>
            <td>11</td>
            <td>{detailMeal.strIngredient1}</td>
            <td>{detailMeal.strMeasure1}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient12
        ? (
          <tr>
            <td>12</td>
            <td>{detailMeal.strIngredient12}</td>
            <td>{detailMeal.strMeasure12}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient13
        ? (
          <tr>
            <td>13</td>
            <td>{detailMeal.strIngredient13}</td>
            <td>{detailMeal.strMeasure13}</td>
          </tr>
        )
        : null}

      {detailMeal.strIngredient14
        ? (
          <tr>
            <td>14</td>
            <td>{detailMeal.strIngredient14}</td>
            <td>{detailMeal.strMeasure14}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient15
        ? (
          <tr>
            <td>15</td>
            <td>{detailMeal.strIngredient15}</td>
            <td>{detailMeal.strMeasure15}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient16
        ? (
          <tr>
            <td>16</td>
            <td>{detailMeal.strIngredient16}</td>
            <td>{detailMeal.strMeasure16}</td>
          </tr>
        )
        : null}

      {detailMeal.strIngredient17
        ? (
          <tr>
            <td>17</td>
            <td>{detailMeal.strIngredient17}</td>
            <td>{detailMeal.strMeasure17}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient18
        ? (
          <tr>
            <td>18</td>
            <td>{detailMeal.strIngredient18}</td>
            <td>{detailMeal.strMeasure18}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient19
        ? (
          <tr>
            <td>19</td>
            <td>{detailMeal.strIngredient19}</td>
            <td>{detailMeal.strMeasure19}</td>
          </tr>
        )
        : null}
      {detailMeal.strIngredient20
        ? (
          <tr>
            <td>20</td>
            <td>{detailMeal.strIngredient20}</td>
            <td>{detailMeal.strMeasure20}</td>
          </tr>
        )
        : null}
    </tbody>
  </table>
);

export default Table;

Table.propTypes = {
  detailMeal: PropTypes.instanceOf(Object).isRequired,
};
