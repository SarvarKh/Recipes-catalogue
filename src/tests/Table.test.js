import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Table from '../components/presentation/Table';

const detailMeal = {
  strIngredient1: 'strIngredient1',
  strIngredient2: 'strIngredient2',
  strIngredient3: 'strIngredient3',
  strIngredient4: 'strIngredient4',
  strIngredient5: 'strIngredient5',
  strIngredient6: 'strIngredient6',
  strIngredient7: 'strIngredient7',
  strIngredient8: 'strIngredient8',
  strIngredient9: 'strIngredient9',
  strIngredient10: 'strIngredient10',
  strMeasure1: 'strMeasure1',
  strMeasure2: 'strMeasure2',
  strMeasure3: 'strMeasure3',
  strMeasure4: 'strMeasure4',
  strMeasure5: 'strMeasure5',
  strMeasure6: 'strMeasure6',
  strMeasure7: 'strMeasure7',
  strMeasure8: 'strMeasure8',
  strMeasure9: 'strMeasure9',
  strMeasure0: 'strMeasure0',
  strMeasure10: 'strMeasure10',
};

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><Table detailMeal={detailMeal} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
