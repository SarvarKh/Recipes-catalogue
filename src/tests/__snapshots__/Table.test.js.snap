// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table>
  <tr>
    <th>
      No
    </th>
    <th>
      Ingredients
    </th>
    <th>
      Measure
    </th>
  </tr>
  <tr>
    <td>
      1
    </td>
    <td>
      strIngredient1
    </td>
    <td>
      strMeasure1
    </td>
  </tr>
  <tr>
    <td>
      2
    </td>
    <td>
      strIngredient2
    </td>
    <td>
      strMeasure2
    </td>
  </tr>
  <tr>
    <td>
      3
    </td>
    <td>
      strIngredient3
    </td>
    <td>
      strMeasure3
    </td>
  </tr>
  <tr>
    <td>
      4
    </td>
    <td>
      strIngredient4
    </td>
    <td>
      strMeasure4
    </td>
  </tr>
  <tr>
    <td>
      5
    </td>
    <td>
      strIngredient5
    </td>
    <td>
      strMeasure5
    </td>
  </tr>
  <tr>
    <td>
      6
    </td>
    <td>
      strIngredient6
    </td>
    <td>
      strMeasure6
    </td>
  </tr>
  <tr>
    <td>
      7
    </td>
    <td>
      strIngredient7
    </td>
    <td>
      strMeasure7
    </td>
  </tr>
  <tr>
    <td>
      8
    </td>
    <td>
      strIngredient8
    </td>
    <td>
      strMeasure8
    </td>
  </tr>
  <tr>
    <td>
      9
    </td>
    <td>
      strIngredient9
    </td>
    <td>
      strMeasure9
    </td>
  </tr>
  <tr>
    <td>
      10
    </td>
    <td>
      strIngredient10
    </td>
    <td>
      strMeasure10
    </td>
  </tr>
  
  
  
  
  
  
  
  
  
  
</table>
`;
